protos/transaction.proto

```diff
@@ -68,6 +68,8 @@ message Transaction {
     map<string, bytes> schema_metadata = 3;
     // Key-value pairs to merge with existing config.
     map<string, string> config_upsert_values = 4;
+    // Additional path URIs for data file distribution in multi-path layouts
+    repeated string initial_data_paths = 5;
```

Comment:
Regardless of what we do in the Python API, at transaction level we should have a list of BasePath , not just string data paths. In BasePath we have important fields like is_dataset_root, as well as a user provided name of the path.